File: edentalbackend-main/DentalOffice.Entities/dbEntities/Appointment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace DentalOffice.Entities.dbEntities
{
    public class Appointment
    {
        public int Id { get; set; }
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
        public int DentistId { get; set; }
        public Dentist Dentist { get; set; }
        public int TreatmentId { get; set; }
        public Treatment Treatment { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }

        [NotMapped]
        public string ClientFullName { get; set; }
        [NotMapped]
        public string DentistFullName { get; set; }
        [NotMapped]
        public string TreatmentName { get; set; }
    }
}

File: eDental-flutter-main/lib/models/appointment.dart
import 'package:dart_json_mapper/dart_json_mapper.dart';
import 'package:edental/models/dentist.dart';
import 'package:edental/models/treatment.dart';
import 'package:edental/models/user.dart';
import 'package:flutter/material.dart';

import '../helpers/dateJsonFormatter.dart';

@jsonSerializable
class Appointment {
  int id;
  @JsonProperty(converter: CustomDateTimeConverter())
  DateTime start;
  @JsonProperty(converter: CustomDateTimeConverter())
  DateTime end;
  int dentistId;
  int treatmentId;
  int userId;
  String? clientFullName;
  String? dentistFullName;
  String? treatmentName;
  Appointment(
      this.id,
      this.start,
      this.end,
      this.dentistId,
      this.treatmentId,
      this.userId,
      this.clientFullName,
      this.dentistFullName,
      this.treatmentName);
}
